diff --git a/media/gpu/sandbox/BUILD.gn b/media/gpu/sandbox/BUILD.gn
index cfcb7fa..1489642 100644
--- a/media/gpu/sandbox/BUILD.gn
+++ b/media/gpu/sandbox/BUILD.gn
@@ -30,7 +30,7 @@ source_set("sandbox") {
   if (use_v4l2_codec) {
     deps += [ "//media/gpu/v4l2" ]
   }
-  if (current_cpu != "s390x" && current_cpu != "ppc64" && is_linux &&
+  if (current_cpu != "s390x" && current_cpu != "ppc64" && ozone_platform_x11 && is_linux &&
       !is_castos) {
     # For DRI_DRIVER_DIR.
     configs += [ "//build/config/linux/dri" ]
