diff --git a/ui/gl/BUILD.gn b/ui/gl/BUILD.gn
index 7998d2d..c7c0042 100644
--- a/ui/gl/BUILD.gn
+++ b/ui/gl/BUILD.gn
@@ -248,6 +248,15 @@ component("gl") {
         data_deps += [ "//third_party/vulkan-deps/vulkan-loader/src:libvulkan" ]
       }
     }
+    if(ozone_platform_wayland) {
+      defines += [ "WAYLAND_GBM" ]
+
+      deps += [
+        "//third_party/minigbm",
+        "//ui/gfx:memory_buffer",
+        "//ui/gfx/linux:gbm",
+      ]
+   }
   }
 
   if (ozone_platform_x11) {
