From 71cbeb22c3d85a838a033ec9f65ced56de432f3a Mon Sep 17 00:00:00 2001
From: qizhang <jony.zhang@nxp.com>
Date: Tue, 12 Mar 2024 23:21:11 +0800
Subject: [PATCH] User Space: enable ethercat_userspace

Signed-off-by: qizhang <jony.zhang@nxp.com>
---
 tool/Makefile.am | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tool/Makefile.am b/tool/Makefile.am
index 69bb1346..f865be94 100644
--- a/tool/Makefile.am
+++ b/tool/Makefile.am
@@ -33,9 +33,9 @@
 
 EXTRA_DIST =
 
-bin_PROGRAMS = ethercat
+bin_PROGRAMS = ethercat_userspace
 
-ethercat_SOURCES = \
+ethercat_userspace_SOURCES = \
 	soe_errors.c \
 	Command.cpp \
 	CommandAlias.cpp \
@@ -73,7 +73,7 @@ ethercat_SOURCES = \
 	sii_crc.cpp
 
 if ENABLE_EOE
-ethercat_SOURCES += CommandEoe.cpp
+ethercat_userspace_SOURCES += CommandEoe.cpp
 else
 EXTRA_DIST += CommandEoe.cpp
 endif
@@ -125,7 +125,7 @@ REV = `if test -s $(top_srcdir)/revision; then \
 		hg id -i $(top_srcdir) 2>/dev/null || echo "unknown"; \
 	fi`
 
-ethercat_CXXFLAGS = \
+ethercat_userspace_CXXFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/master \
 	-DEC_MASTER_IN_USERSPACE \
-- 
2.25.1

