From dfb044af9689ef2d95c30a6b13eb9d306ce0a732 Mon Sep 17 00:00:00 2001
From: qizhang <jony.zhang@nxp.com>
Date: Tue, 30 Apr 2024 23:32:43 +0800
Subject: [PATCH] fix compile error because of linux kernel upgrade to v6.6.3

Signed-off-by: qizhang <jony.zhang@nxp.com>
Upstream-Status: Pending
---
 master/cdev.c   | 2 +-
 master/module.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/master/cdev.c b/master/cdev.c
index 7ded08e..d16e405 100644
--- a/master/cdev.c
+++ b/master/cdev.c
@@ -235,7 +235,7 @@ int eccdev_mmap(
     EC_MASTER_DBG(priv->cdev->master, 1, "mmap()\n");
 
     vma->vm_ops = &eccdev_vm_ops;
-    vma->vm_flags |= VM_DONTDUMP; /* Pages will not be swapped out */
+    vm_flags_set(vma, VM_DONTDUMP);
     vma->vm_private_data = priv;
 
     return 0;
diff --git a/master/module.c b/master/module.c
index 573a694..4f8170b 100644
--- a/master/module.c
+++ b/master/module.c
@@ -112,7 +112,7 @@ int __init ec_init_module(void)
         }
     }
 
-    class = class_create(THIS_MODULE, "EtherCAT");
+    class = class_create("EtherCAT");
     if (IS_ERR(class)) {
         EC_ERR("Failed to create device class.\n");
         ret = PTR_ERR(class);
-- 
2.25.1

