From 55b488bc9fc54c7d2f58a884dfa2b409e4d0b29f Mon Sep 17 00:00:00 2001
From: Hongbo Wang <hongbo.wang@nxp.com>
Date: Tue, 19 Mar 2024 11:14:41 +0800
Subject: [PATCH 1/5] Update configure.ac to avoid compile error

Signed-off-by: Hongbo Wang <hongbo.wang@nxp.com>
---
 configure.ac | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/configure.ac b/configure.ac
index 18737f02..c38bb2ff 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,6 +93,12 @@ AC_SUBST(ENABLE_USERSPACE_MASTER,[$enable_usecat])
 # Linux sources
 #------------------------------------------------------------------------------
 
+if test "x${enable_usecat}" = "x1"; then
+	version=[`uname -r`]
+	modulesdir=/lib/modules/${version}
+	sourcedir=`cd ${modulesdir}/build && pwd -P`
+	AC_SUBST(LINUX_SOURCE_DIR,[$sourcedir])
+else
 AC_ARG_WITH([linux-dir],
     AC_HELP_STRING(
         [--with-linux-dir=<DIR>],
@@ -148,6 +154,7 @@ linuxversion=`echo $kernelrelease | grep -oE "$regex"`
 
 AC_SUBST(LINUX_SOURCE_DIR,[$sourcedir])
 AC_MSG_RESULT([$LINUX_SOURCE_DIR (Kernel $linuxversion)])
+fi
 
 #------------------------------------------------------------------------------
 # Linux module installation subdirectory
-- 
2.34.1

