From 1053cb718ec961e0fb3d9c1858695858edef38f8 Mon Sep 17 00:00:00 2001
From: Hongbo Wang <hongbo.wang@nxp.com>
Date: Thu, 9 May 2024 06:50:19 +0000
Subject: [PATCH 4/5] Add support for platform with 1 core, like i.MX91

Signed-off-by: Hongbo Wang <hongbo.wang@nxp.com>
---
 devices/enet_fec/enet_main.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/devices/enet_fec/enet_main.c b/devices/enet_fec/enet_main.c
index 25e4efbc..c0de1f05 100644
--- a/devices/enet_fec/enet_main.c
+++ b/devices/enet_fec/enet_main.c
@@ -11,6 +11,7 @@
 #include <linux/netdevice.h>
 #include <linux/if_arp.h> /* ARPHRD_ETHER */
 #include <linux/etherdevice.h>
+#include <sys/sysinfo.h>
 
 #include "../globals.h"
 #include "../ecdev.h"
@@ -60,17 +61,26 @@ int ecus_bind_cpu_core(uint32_t core_mask)
 {
     pthread_t thread;
     cpu_set_t cpuset;
+    int core_num = 0;
     int ret = 0;
     int i = 0;
 
     thread = pthread_self();
+    core_num = get_nprocs();
 
     CPU_ZERO(&cpuset);
+
+    if (core_num == 1) {
+        CPU_SET(0, &cpuset);
+        goto set_tag;
+    }
+
     for (i = 0; i < 32; i++) {
         if (core_mask & (0x01 << i))
             CPU_SET(i, &cpuset);
     }
 
+set_tag:
     ret = pthread_setaffinity_np(thread, sizeof(cpu_set_t), &cpuset);
     if (ret != 0)
         printk(KERN_ERR PFX "pthread_setaffinity_np failed!\n");
-- 
2.34.1

