From a5f4f775f08048ec4f9cb03cc7476c355197b2cf Mon Sep 17 00:00:00 2001
From: Jony Zhang <jony.zhang@nxp.com>
Date: Fri, 8 May 2026 19:58:59 +0800
Subject: [PATCH 6/6] examples: mini: fix compile error because of kernel
 upgrade to 6.18.2

Signed-off-by: Jony Zhang <jony.zhang@nxp.com>
Upstream-Status: Pending
---
 examples/mini/mini.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/mini/mini.c b/examples/mini/mini.c
index 8c854e82..9d80d17d 100644
--- a/examples/mini/mini.c
+++ b/examples/mini/mini.c
@@ -525,7 +525,7 @@ void __exit cleanup_mini_module(void)
 {
     printk(KERN_INFO PFX "Stopping...\n");
 
-    del_timer_sync(&timer);
+    timer_delete_sync(&timer);
 
 #if EXTERNAL_MEMORY
     kfree(domain1_pd);
-- 
2.43.0

