From d6b47fb315bba0e434f17feebc364f479aac1d38 Mon Sep 17 00:00:00 2001
From: "hongbo.wang" <hongbo.wang@nxp.com>
Date: Sat, 27 Feb 2021 17:58:55 +0800
Subject: [PATCH 01/22] configs: ls1043a-rdb: add fman ucode memory for root
 cell

Signed-off-by: hongbo.wang <hongbo.wang@nxp.com>
---
 configs/arm64/ls1043a-rdb.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/configs/arm64/ls1043a-rdb.c b/configs/arm64/ls1043a-rdb.c
index e1d3c48e..ab720876 100644
--- a/configs/arm64/ls1043a-rdb.c
+++ b/configs/arm64/ls1043a-rdb.c
@@ -1,7 +1,7 @@
 /*
  * ls1043a RDB target - linux-demo
  *
- * Copyright 2020 NXP
+ * Copyright 2020-2021 NXP
  *
  * Authors:
  *  Hongbo Wang <hongbo.wang@nxp.com>
@@ -16,7 +16,7 @@
 struct {
 	struct jailhouse_system header;
 	__u64 cpus[1];
-	struct jailhouse_memory mem_regions[61];
+	struct jailhouse_memory mem_regions[62];
 	struct jailhouse_irqchip irqchips[2];
 	struct jailhouse_pci_device pci_devices[2];
 } __attribute__((packed)) config = {
@@ -119,6 +119,13 @@ struct {
 			.flags = JAILHOUSE_MEM_READ | JAILHOUSE_MEM_WRITE |
 				JAILHOUSE_MEM_EXECUTE,
 		},
+		/* RAM - FMAN ucode - root cell */ {
+			.phys_start = 0xf4000000,
+			.virt_start = 0xf4000000,
+			.size = 0xc000000,
+			.flags = JAILHOUSE_MEM_READ | JAILHOUSE_MEM_WRITE |
+				JAILHOUSE_MEM_EXECUTE,
+		},
 		/* RAM: loader */ {
 			.phys_start = 0xc0400000,
 			.virt_start = 0xc0400000,
-- 
2.43.0

