From c606dfe7cedb70d0db523ef40a231da783ba36fc Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Mon, 20 Feb 2023 16:09:31 +0800
Subject: [PATCH 4/6] lldpd: preemption: fix status not updated issue

Signed-off-by: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
---
 src/daemon/protocols/lldp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/daemon/protocols/lldp.c b/src/daemon/protocols/lldp.c
index e8e0207..ef3da4e 100644
--- a/src/daemon/protocols/lldp.c
+++ b/src/daemon/protocols/lldp.c
@@ -999,8 +999,9 @@ lldp_decode(struct lldpd *cfg, char *frame, int s, struct lldpd_hardware *hardwa
 						ioctl(cfg->g_sock, SIOCETHTOOL, &ifr);
 
 						if (!port->p_fp.active) {
-							lldpd_update_localports(cfg);
+							lport->p_fp.active = 1;
 							lldp_send(cfg, hardware);
+							lldpd_update_localports(cfg);
 						}
 					}
 					break;
-- 
2.25.1

