From e3d0b68e2e17b0e4b06401a04f42f4aaaaa876b9 Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Tue, 2 Nov 2021 11:16:26 +0800
Subject: [PATCH 4/4] ethtool: Add configure to compile preempt file

Signed-off-by: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
---
 Makefile.in | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index fa26847..afba2f6 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -108,7 +108,7 @@ sbin_PROGRAMS = ethtool$(EXEEXT)
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/channels.c netlink/coalesce.c netlink/pause.c \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/eee.c netlink/tsinfo.c \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/desc-ethtool.c netlink/desc-genlctrl.c \
-@ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/desc-rtnl.c netlink/cable_test.c netlink/tunnels.c \
+@ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/desc-rtnl.c netlink/cable_test.c netlink/tunnels.c netlink/preempt.c \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  uapi/linux/ethtool_netlink.h \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  uapi/linux/netlink.h uapi/linux/genetlink.h \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  uapi/linux/rtnetlink.h uapi/linux/if_link.h
@@ -155,8 +155,8 @@ am__ethtool_SOURCES_DIST = ethtool.c uapi/linux/ethtool.h internal.h \
 	netlink/rings.c netlink/channels.c netlink/coalesce.c \
 	netlink/pause.c netlink/eee.c netlink/tsinfo.c \
 	netlink/desc-ethtool.c netlink/desc-genlctrl.c \
-	netlink/desc-rtnl.c netlink/cable_test.c netlink/tunnels.c \
-	uapi/linux/ethtool_netlink.h uapi/linux/netlink.h \
+	netlink/desc-rtnl.c netlink/cable_test.c netlink/preempt.c netlink/tunnels.c \
+	uapi/linux/ethtool_netlink.h uapi/linux/netlink.h uapi/linux/ethtool_netlink.h \
 	uapi/linux/genetlink.h uapi/linux/rtnetlink.h \
 	uapi/linux/if_link.h
 @ETHTOOL_ENABLE_PRETTY_DUMP_TRUE@am__objects_1 = amd8111e.$(OBJEXT) \
@@ -241,7 +241,7 @@ am__test_cmdline_SOURCES_DIST = test-cmdline.c test-common.c ethtool.c \
 	netlink/channels.c netlink/coalesce.c netlink/pause.c \
 	netlink/eee.c netlink/tsinfo.c netlink/desc-ethtool.c \
 	netlink/desc-genlctrl.c netlink/desc-rtnl.c \
-	netlink/cable_test.c netlink/tunnels.c \
+	netlink/cable_test.c netlink/tunnels.c netlink/preempt.c uapi/linux/ethtool_netlink.h \
 	uapi/linux/ethtool_netlink.h uapi/linux/netlink.h \
 	uapi/linux/genetlink.h uapi/linux/rtnetlink.h \
 	uapi/linux/if_link.h
@@ -334,6 +334,7 @@ am__test_features_SOURCES_DIST = test-features.c test-common.c \
 	netlink/pause.c netlink/eee.c netlink/tsinfo.c \
 	netlink/desc-ethtool.c netlink/desc-genlctrl.c \
 	netlink/desc-rtnl.c netlink/cable_test.c netlink/tunnels.c \
+	netlink/preempt.c uapi/linux/ethtool_netlink.h \
 	uapi/linux/ethtool_netlink.h uapi/linux/netlink.h \
 	uapi/linux/genetlink.h uapi/linux/rtnetlink.h \
 	uapi/linux/if_link.h
-- 
2.27.0

