From c1499a26cff53ff17faddf501b7f44161d571b9e Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Mon, 7 Feb 2022 15:45:41 +0800
Subject: [PATCH 4/4] ethtool: Add configure to compile preempt file

Signed-off-by: Changming Huang <jerry.huang@nxp.com>
---
 Makefile.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index c849d51..b5445f8 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -110,7 +110,7 @@ sbin_PROGRAMS = ethtool$(EXEEXT)
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/stats.c \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/desc-ethtool.c netlink/desc-genlctrl.c \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/module-eeprom.c \
-@ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/desc-rtnl.c netlink/cable_test.c netlink/tunnels.c \
+@ETHTOOL_ENABLE_NETLINK_TRUE@		  netlink/desc-rtnl.c netlink/cable_test.c netlink/tunnels.c netlink/preempt.c \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  uapi/linux/ethtool_netlink.h \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  uapi/linux/netlink.h uapi/linux/genetlink.h \
 @ETHTOOL_ENABLE_NETLINK_TRUE@		  uapi/linux/rtnetlink.h uapi/linux/if_link.h
@@ -158,7 +158,7 @@ am__ethtool_SOURCES_DIST = ethtool.c uapi/linux/ethtool.h internal.h \
 	netlink/pause.c netlink/eee.c netlink/tsinfo.c netlink/fec.c \
 	netlink/stats.c netlink/desc-ethtool.c netlink/desc-genlctrl.c \
 	netlink/module-eeprom.c netlink/desc-rtnl.c \
-	netlink/cable_test.c netlink/tunnels.c \
+	netlink/cable_test.c netlink/tunnels.c netlink/preempt.c \
 	uapi/linux/ethtool_netlink.h uapi/linux/netlink.h \
 	uapi/linux/genetlink.h uapi/linux/rtnetlink.h \
 	uapi/linux/if_link.h
@@ -247,7 +247,7 @@ am__test_cmdline_SOURCES_DIST = test-cmdline.c test-common.c ethtool.c \
 	netlink/eee.c netlink/tsinfo.c netlink/fec.c netlink/stats.c \
 	netlink/desc-ethtool.c netlink/desc-genlctrl.c \
 	netlink/module-eeprom.c netlink/desc-rtnl.c \
-	netlink/cable_test.c netlink/tunnels.c \
+	netlink/cable_test.c netlink/tunnels.c netlink/preempt.c \
 	uapi/linux/ethtool_netlink.h uapi/linux/netlink.h \
 	uapi/linux/genetlink.h uapi/linux/rtnetlink.h \
 	uapi/linux/if_link.h
@@ -343,7 +343,7 @@ am__test_features_SOURCES_DIST = test-features.c test-common.c \
 	netlink/pause.c netlink/eee.c netlink/tsinfo.c netlink/fec.c \
 	netlink/stats.c netlink/desc-ethtool.c netlink/desc-genlctrl.c \
 	netlink/module-eeprom.c netlink/desc-rtnl.c \
-	netlink/cable_test.c netlink/tunnels.c \
+	netlink/cable_test.c netlink/tunnels.c netlink/preempt.c \
 	uapi/linux/ethtool_netlink.h uapi/linux/netlink.h \
 	uapi/linux/genetlink.h uapi/linux/rtnetlink.h \
 	uapi/linux/if_link.h
-- 
2.17.1

