diff --git a/scripts/remove.sh b/scripts/remove.sh
index 5f3ec06..f9bcd34 100755
--- a/scripts/remove.sh
+++ b/scripts/remove.sh
@@ -27,10 +27,30 @@ MODULES=(
 "ietf-tls-server@2019-07-02.yang -e local-client-auth-supported"
 "ietf-netconf-server@2019-07-02.yang -e ssh-listen -e tls-listen -e ssh-call-home -e tls-call-home"
 "ietf-interfaces@2018-02-20.yang"
-"ietf-ip@2018-02-22.yang"
+"ietf-ip@2018-02-22.yang -e ipv4-non-contiguous-netmasks"
 "ietf-network-instance@2019-01-21.yang"
 "ietf-subscribed-notifications@2019-09-09.yang -e encode-xml -e replay -e subtree -e xpath"
 "ietf-yang-push@2019-09-09.yang -e on-change"
+"iana-if-type@2023-01-26.yang"
+"ieee802-types@2023-10-22.yang"
+"ieee802-dot1q-types@2023-10-26.yang"
+"ieee802-dot1q-qci-augment@2024-12-18.yang"
+"ieee802-dot1q-pb@2023-10-22.yang"
+"ieee802-dot1q-bridge@2023-10-26.yang"
+"ieee802-dot1cb-frer-types@2021-12-08.yang"
+"ieee802-dot1cb-frer@2021-12-08.yang"
+"ieee802-dot1cb-stream-identification-types@2021-12-09.yang"
+"ieee802-dot1cb-stream-identification@2021-12-08.yang"
+"ieee802-dot1q-preemption@2023-10-26.yang -e frame-preemption"
+"ieee802-dot1q-preemption-bridge@2023-10-26.yang"
+"ieee802-dot1q-psfp@2023-10-26.yang -e psfp"
+"ieee802-dot1q-psfp-bridge@2023-10-26.yang"
+"ieee802-dot1q-sched@2023-10-22.yang -e scheduled-traffic"
+"ieee802-dot1q-sched-bridge@2023-10-26.yang"
+"ieee802-dot1q-stream-filters-gates@2023-07-03.yang -e closed-gate-state"
+"ieee802-dot1q-stream-filters-gates-bridge@2023-07-03.yang"
+"ietf-system@2014-08-06.yang -e authentication -e local-users"
+"nxp-bridge-vlan-tc-flower@2020-04-02.yang"
 )
 
 CMD_UNINSTALL=
diff --git a/scripts/setup.sh b/scripts/setup.sh
index 45ec2c7..86387ad 100755
--- a/scripts/setup.sh
+++ b/scripts/setup.sh
@@ -36,10 +36,30 @@ MODULES=(
 "ietf-tls-server@2019-07-02.yang -e local-client-auth-supported"
 "ietf-netconf-server@2019-07-02.yang -e ssh-listen -e tls-listen -e ssh-call-home -e tls-call-home"
 "ietf-interfaces@2018-02-20.yang"
-"ietf-ip@2018-02-22.yang"
+"ietf-ip@2018-02-22.yang -e ipv4-non-contiguous-netmasks"
 "ietf-network-instance@2019-01-21.yang"
 "ietf-subscribed-notifications@2019-09-09.yang -e encode-xml -e replay -e subtree -e xpath"
 "ietf-yang-push@2019-09-09.yang -e on-change"
+"iana-if-type@2023-01-26.yang"
+"ieee802-types@2023-10-22.yang"
+"ieee802-dot1q-types@2023-10-26.yang"
+"ieee802-dot1q-qci-augment@2024-12-18.yang"
+"ieee802-dot1q-pb@2023-10-22.yang"
+"ieee802-dot1q-bridge@2023-10-26.yang"
+"ieee802-dot1cb-frer-types@2021-12-08.yang"
+"ieee802-dot1cb-frer@2021-12-08.yang"
+"ieee802-dot1cb-stream-identification-types@2021-12-09.yang"
+"ieee802-dot1cb-stream-identification@2021-12-08.yang"
+"ieee802-dot1q-preemption@2023-10-26.yang -e frame-preemption"
+"ieee802-dot1q-preemption-bridge@2023-10-26.yang"
+"ieee802-dot1q-psfp@2023-10-26.yang -e psfp"
+"ieee802-dot1q-psfp-bridge@2023-10-26.yang"
+"ieee802-dot1q-sched@2023-10-22.yang -e scheduled-traffic"
+"ieee802-dot1q-sched-bridge@2023-10-26.yang"
+"ieee802-dot1q-stream-filters-gates@2023-07-03.yang -e closed-gate-state"
+"ieee802-dot1q-stream-filters-gates-bridge@2023-07-03.yang"
+"ietf-system@2014-08-06.yang -e authentication -e local-users"
+"nxp-bridge-vlan-tc-flower@2020-04-02.yang"
 )
 
 CMD_INSTALL=
