From 7e9688de7c5b642cd1eb688eb617b4db52785887 Mon Sep 17 00:00:00 2001
From: Nikhil Gupta <nikhil.gupta_2@nxp.com>
Date: Mon, 13 May 2024 02:50:27 -0700
Subject: [PATCH] la9310:Correct the PCI_CRTL_ADDR in M7

Signed-off-by: Nikhil Gupta <nikhil.gupta_2@nxp.com>
BSP5G-1752
---
 rfnm_m7.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/rfnm_m7.h b/rfnm_m7.h
index 4875048..6576c1c 100644
--- a/rfnm_m7.h
+++ b/rfnm_m7.h
@@ -1,8 +1,8 @@
 #ifndef __H_RFNM_M7
 #define __H_RFNM_M7
 
-#define PCIE_CTRL_ADDR_OFFSET (0x1C00F740 - 0x18000000)
-#define PCIE_CTRL_ADDR_FROM_M7 (0xD0000000 + PCIE_CTRL_ADDR_OFFSET) // D400 F740
+#define PCIE_CTRL_ADDR_OFFSET (0x1C010B40 - 0x18000000)
+#define PCIE_CTRL_ADDR_FROM_M7 (0xD0000000 + PCIE_CTRL_ADDR_OFFSET) // D400 10B40
 #define PCIE_GPIO_ADDR_FROM_M7 (0xD0000000 + 0x2300000)
 
 #define INTERRUPT_LATENCY_OFFSET (5.5 * 3.84)
@@ -50,4 +50,4 @@ void rfnm_fe_trigger_latches(struct rfnm_m7_dgb * dgb_dt, int txrx);
 void rfnm_fe_manual_clock(int dgb_id, int id);
 int rfnm_fe_generic_init(struct rfnm_m7_dgb * dgb_dt);
 
-#endif
\ No newline at end of file
+#endif
-- 
2.25.1

